### PR TITLE
Release 2.0.3

### PR DESCRIPTION
**Pull request by mcriscolo-cs**
# Checklist:
Please check you fulfill ALL of the relevant checkboxes
- [ ] Notified docs of any potential USER-facing changes
- [ ] Added short description of the change - with relevant motivation and context. 
- [ ] Branch has the ticket number in its name (along with a ticket summary)
- [ ] Commented the code, particularly in hard-to-understand areas
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Ran `npm test` and it passes
- [ ] Changes generate no new warnings

### CODE REVIEW

File: src/commands/assets/templates/skill/daemon/README.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ Note:
 * `Dockerfile` to build Docker image for this skill
 
 #### Steps
+
+A Makefile is provided to do these steps. Set environment variables `DOCKER_PREGISTRY_URL` (like <docker-registry-url>/<namespace-org>) and `PROJECT_NAME` (Cortex Project Name) and use Makefile to deploy skill.
+`make all` will build & push Docker image, deploy Cortex Action and Skill, and then invoke skill to test.
```
Comment (smichalski-cs) [inline src/commands/assets/templates/skill/daemon/README.md:18]: use "and" not "&".  Capitalize Skill after invoke.

File: src/commands/assets/templates/skill/external-api/README.md [hunk 2]
```diff
@@ -5,6 +5,9 @@ This skill type allows users to wrap an external REST API as a Cortex Fabric Ski
 1. Update `skill.yaml` with `HTTP Method`, `URL`, `Path` and `headers` per the targeted external API
 2. Deploy the Skill `cortex skills save -y skill.yaml --project <Project Name>`
 
+A Makefile is provided to do these steps. Set environment variable `PROJECT_NAME` (Cortex Project Name) and use Makefile to deploy skill.
+`make all` will deploy Cortex Skill, and then invoke skill to test.
```
Comment (smichalski-cs) [inline src/commands/assets/templates/skill/external-api/README.md:9]: Capitalize Skill after invoke

File: src/commands/assets/templates/skill/job/README.md [hunk 3]
```diff
@@ -9,6 +9,10 @@ Cortex skill that runs a background job.
 * `Dockerfile` to build Docker image for this skill
 
 #### Steps
+
+A Makefile is provided to do these steps. Set environment variables `DOCKER_PREGISTRY_URL` (like <docker-registry-url>/<namespace-org>) and `PROJECT_NAME` (Cortex Project Name) and use Makefile to deploy skill.
+`make all` will build & push Docker image, deploy Cortex Action and Skill, and then invoke skill to test.
```
Comment (smichalski-cs) [inline src/commands/assets/templates/skill/job/README.md:14]: use "and" not "&". Capitalize Skill after invoke.

File: src/commands/assets/templates/skill/job/README.md [hunk 4]
```diff
@@ -9,6 +9,10 @@ Cortex skill that runs a background job.
 * `Dockerfile` to build Docker image for this skill
 
 #### Steps
+
+A Makefile is provided to do these steps. Set environment variables `DOCKER_PREGISTRY_URL` (like <docker-registry-url>/<namespace-org>) and `PROJECT_NAME` (Cortex Project Name) and use Makefile to deploy skill.
```
Comment (smichalski-cs) [inline src/commands/assets/templates/skill/job/README.md:13]: Capitalize Skill

File: src/commands/assets/templates/skill/daemon/README.md [hunk 5]
```diff
@@ -13,6 +13,10 @@ Note:
 * `Dockerfile` to build Docker image for this skill
 
 #### Steps
+
+A Makefile is provided to do these steps. Set environment variables `DOCKER_PREGISTRY_URL` (like <docker-registry-url>/<namespace-org>) and `PROJECT_NAME` (Cortex Project Name) and use Makefile to deploy skill.
```
Comment (smichalski-cs) [inline src/commands/assets/templates/skill/daemon/README.md:17]: Capitalize Skill

File: src/commands/plopfile.js [hunk 6]
```diff
@@ -7,7 +7,8 @@ module.exports = (plop) => {
             {
                 type: 'input',
                 name: 'name',
-                message: 'Skill Name?',
+                message: 'Skill Name? (Must be alphanumeric)',
```
Comment (smichalski-cs) [inline src/commands/plopfile.js:10]: "Skill name must be alphanumeric, beginning with a letter. Hyphen and underscore allowed internally."

File: src/commands/skills.js [hunk 7]
```diff
@@ -117,6 +114,51 @@ module.exports.DescribeSkillCommand = class DescribeSkillCommand {
...
+    execute(skillName, options) {
+        const profile = loadProfile(options.profile);
+        debug('%s.executeDeploySkill(%s)', profile.name, skillName);
+
+        const catalog = new Catalog(profile.url);
+        catalog.deploySkill(options.project || profile.project, profile.token, skillName, options.verbose).then((response) => {
+            if (response.success) {
+                printSuccess(`Deployed skill ${skillName}: ${response.message}`, options);
+            } else {
+                printError(`Failed to deploy skill ${skillName}: ${response.message}`, options);
```
Comment (smichalski-cs) [inline src/commands/skills.js:153]: Capitalize Skill

File: src/commands/skills.js [hunk 8]
```diff
@@ -117,6 +114,51 @@ module.exports.DescribeSkillCommand = class DescribeSkillCommand {
...
+            }
+        })
+            .catch((err) => {
+                printError(`Failed to deploy skill ${skillName}: ${err.status} ${err.message}`, options);
```
Comment (smichalski-cs) [inline src/commands/skills.js:157]: Capitalize Skill

File: src/commands/skills.js [hunk 9]
```diff
@@ -117,6 +114,51 @@ module.exports.DescribeSkillCommand = class DescribeSkillCommand {
...
+    }
+
+    execute(skillName, options) {
+        const profile = loadProfile(options.profile);
+        debug('%s.executeDeploySkill(%s)', profile.name, skillName);
+
+        const catalog = new Catalog(profile.url);
+        catalog.deploySkill(options.project || profile.project, profile.token, skillName, options.verbose).then((response) => {
+            if (response.success) {
+                printSuccess(`Deployed skill ${skillName}: ${response.message}`, options);
```
Comment (smichalski-cs) [inline src/commands/skills.js:151]: Capitalize Skill

File: src/commands/skills.js (hunk 7 above)
Comment (smichalski-cs) [inline src/commands/skills.js:153]: Capitalize Skill

File: src/commands/skills.js [hunk 10]
```diff
@@ -117,6 +114,51 @@ module.exports.DescribeSkillCommand = class DescribeSkillCommand {
...
+
+    execute(skillName, options) {
+        const profile = loadProfile(options.profile);
+        debug('%s.executeUndeploySkill(%s)', profile.name, skillName);
+        const catalog = new Catalog(profile.url);
+        catalog.unDeploySkill(options.project || profile.project, profile.token, skillName, options.verbose).then((response) => {
+            if (response.success) {
+                printSuccess(`Undeploy skill ${skillName}: ${response.message}`, options);
+            } else {
+                printError(`Failed to Undeploy skill ${skillName}: ${response.message}`, options);
```
Comment (smichalski-cs) [inline src/commands/skills.js:130]: capitalize Skill

Comment (smichalski-cs) [inline src/commands/skills.js:130]: capitalize Skill

File: src/commands/skills.js [hunk 11]
```diff
@@ -117,6 +114,51 @@ module.exports.DescribeSkillCommand = class DescribeSkillCommand {
...
+            }
+        })
+            .catch((err) => {
+                printError(`Failed to Undeploy skill ${skillName}: ${err.status} ${err.message}`, options);
```
Comment (smichalski-cs) [inline src/commands/skills.js:134]: capitalize Skill

**Review by smichalski-cs — CHANGES_REQUESTED** [review]

**Review by vkarumuri-cs — APPROVED** [review]

File: src/commands/skills.js (hunk 11 above)
Comment (mcriscolo-cs) [reply src/commands/skills.js]: fixed

File: src/commands/skills.js (hunk 10 above)
Comment (mcriscolo-cs) [reply src/commands/skills.js]: fixed

Comment (mcriscolo-cs) [reply src/commands/skills.js]: fixed

File: src/commands/skills.js (hunk 7 above)
Comment (mcriscolo-cs) [reply src/commands/skills.js]: fixed

File: src/commands/skills.js (hunk 9 above)
Comment (mcriscolo-cs) [reply src/commands/skills.js]: fixed

File: src/commands/assets/templates/skill/daemon/README.md (hunk 1 above)
Comment (mcriscolo-cs) [reply src/commands/assets/templates/skill/daemon/README.md]: fixed

File: bin/cortex-secrets.js [hunk 12]
```diff
@@ -22,51 +22,52 @@ const program = require('../src/commander');
...
             console.error(chalk.red(err.message));
         }
     }));
 
-// Write Secure Variable Value
+// Write Secure Value
 program
     .command('save <keyName> [value]')
-    .description('Save or overwrite a secure variable value. By default, values are stored as strings but can also be saved as JSON or YAML values.')
+    .description('Save or overwrite a secure value. By default, values are stored as strings but can also be saved as JSON or YAML values.')
```
[the remark this answers is not in the document]
Comment (mcriscolo-cs) [reply bin/cortex-secrets.js]: fixed

File: bin/cortex-docker.js [hunk 13]
```diff
@@ -35,6 +35,7 @@ program
     .option('--color [on/off]', 'Turn on/off colors for JSON output.', 'on')
     .option('--profile [profile]', 'The profile to use')
     .option('--project [project]', 'The project to use')
+    .option('--ttl [time]', 'The amount of time for this login to remain active (ex. "14d")', '14d')
```
[the remark this answers is not in the document]
Comment (mcriscolo-cs) [reply bin/cortex-docker.js]: fixed

File: bin/cortex-configure.js [hunk 14]
```diff
@@ -47,13 +47,10 @@ program
     .option('--profile [profile]', 'The profile to use')
     .option('--project [project]', 'The project to use')
     .option('--color [on/off]', 'Turn on/off colors for JSON output.', 'on')
-    .action(() => {
+    .option('--ttl [time]', 'The amount of time for this login to remain active (ex. "1d")', '1d')
```
[the remark this answers is not in the document]
Comment (mcriscolo-cs) [reply bin/cortex-configure.js]: fixed

File: src/commands/assets/templates/skill/external-api/README.md (hunk 2 above)
Comment (mcriscolo-cs) [reply src/commands/assets/templates/skill/external-api/README.md]: fixed

File: src/commands/assets/templates/skill/job/README.md (hunk 3 above)
Comment (mcriscolo-cs) [reply src/commands/assets/templates/skill/job/README.md]: fixed

File: src/commands/assets/templates/skill/daemon/README.md (hunk 5 above)
Comment (mcriscolo-cs) [reply src/commands/assets/templates/skill/daemon/README.md]: fixed

File: src/commands/skills.js (hunk 7 above)
Comment (mcriscolo-cs) [reply src/commands/skills.js]: fixed

File: src/commands/skills.js (hunk 8 above)
Comment (mcriscolo-cs) [reply src/commands/skills.js]: fixed

File: src/commands/assets/templates/skill/job/README.md (hunk 4 above)
Comment (mcriscolo-cs) [reply src/commands/assets/templates/skill/job/README.md]: fixed

File: src/commands/plopfile.js (hunk 6 above)
Comment (mcriscolo-cs) [reply src/commands/plopfile.js]: fixed

**Review by smichalski-cs — APPROVED** [review]

**Review by rsrivastava-cs — APPROVED** [review]

**Pull request merged**